device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -468,9 +470,29 @@ public void onAuthenticationSessionOpened()
     }
 
     @Override
-    public void onMessageAcknowledged(Message message)
+    public void onMessageAcknowledged(Message message, DeliveryState deliveryState)
     {
-        this.listener.onMessageSent(message, null);
+        if (deliveryState == Accepted.getInstance())
+        {
+            this.listener.onMessageSent(message, null);
+        }
+        else if (deliveryState instanceof Rejected)
```

Comment:
why is the check for rejected look different than the other delivery states? 
Shouldn't it be 
`deliveryState == Rejected.getInstance()` ?

Reply:
Rejected is the one delivery state that comes with additional information in the ack. For Accepted/Released, there is no payload telling why the message was Accepted/Released. For Rejected, there should be some error details explaining why the message was rejected